internal/troubleshoot/collect/config_dump.go

```diff
@@ -0,0 +1,120 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package collect
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"path"
+
+	troubleshootv1b2 "github.com/replicatedhq/troubleshoot/pkg/apis/troubleshoot/v1beta2"
+	tbcollect "github.com/replicatedhq/troubleshoot/pkg/collect"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	kube "github.com/envoyproxy/gateway/internal/kubernetes"
+)
+
+var _ tbcollect.Collector = &ConfigDump{}
+
+// ConfigDump defines a collector that dumps the envoy configuration of the proxy pod.
+type ConfigDump struct {
+	BundlePath   string
+	Namespace    string
+	ClientConfig *rest.Config
+}
+
+func (cd ConfigDump) Title() string {
+	return "config-dump"
+}
+
+func (cd ConfigDump) IsExcluded() (bool, error) {
+	return false, nil
+}
+
+func (cd ConfigDump) GetRBACErrors() []error {
+	return nil
+}
+
+func (cd ConfigDump) HasRBACErrors() bool {
+	return false
+}
+
+func (cd ConfigDump) CheckRBAC(_ context.Context, _ tbcollect.Collector, _ *troubleshootv1b2.Collect, _ *rest.Config, _ string) error {
+	return nil
+}
+
+func (cd ConfigDump) Collect(_ chan<- interface{}) (tbcollect.CollectorResult, error) {
+	client, err := kubernetes.NewForConfig(cd.ClientConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	pods, err := listPods(context.TODO(), client, cd.Namespace, labels.SelectorFromSet(map[string]string{
+		"app.kubernetes.io/component":  "proxy",
+		"app.kubernetes.io/managed-by": "envoy-gateway",
```

Comment:
What if more than one eg instance installed in the same ns, this will dump all proxies under the ns?

Reply:
yes